Added Event report with Report Filter as a seperate component

mail2bishnoi opened the pull request:
Added Event Report with Report Filter as a separate component.

tananaev commented on modern/src/App.js:
```diff
@@ -1,41 +1,51 @@
+import UsersPage from "./admin/UsersPage";
+import DevicePage from "./DevicePage";
+import UserPage from "./UserPage";
+import SocketController from "./SocketController";
```
Please revert code style changes. We use single quotes.

mail2bishnoi replied:
done.

tananaev commented:
My comment applied to all the code. Can you please revert all formatting changes that you did.

mail2bishnoi commented:
Revert all the code formatting applied by the vscode. Can you please share which code format you are using?

tananaev commented:
We don't use automated formatting.

tananaev commented on modern/src/reports/ReportFilter.js:
```diff
@@ -0,0 +1,116 @@
+          variant="contained"
+          disabled={!deviceId}
+          onClick={handleShow}
+        >
```
Closing brackets should be on the previous line. Applies to all places.

mail2bishnoi replied:
done.

tananaev commented on modern/src/reports/ReportFilter.js:
```diff
@@ -0,0 +1,116 @@
+      </FormControl>
+    </>
+  );
+  
```
Remove space

mail2bishnoi replied:
done.

tananaev commented on modern/src/reports/ReportFilter.js:
```diff
@@ -0,0 +1,116 @@
+  
+}
+
+export default ReportFilter;
```
Add line break

mail2bishnoi replied:
done.

tananaev commented on modern/src/reports/ReportFilter.js:
```diff
@@ -0,0 +1,116 @@
+import { useSelector } from 'react-redux';
+import moment from 'moment';
+
+const ReportFilter = (props) => {
```
Can we update route report to use this? Also, how can we add additional filed here? For example for events report we need to filter by event type.

mail2bishnoi replied:
I'll update route report using this.

I think we can use reportType attribute for showing custom fields in the ReportFilter component. Like Below-

<ReportFilter reportType = 'event|route|...' handleSubmit={handleSubmit} />

{ reportType && reportType === 'event' && (
	<FormControl variant="filled" margin="normal" fullWidth>
		<InputLabel>Event Type</InputLabel>
		<Select value={eventType} onChange={(e) => setEventType(e.target.value)}>
			<MenuItem value="ignitionOn">{t('eventIgnitionOn')}</MenuItem>
			<MenuItem value="ignitionOff">{t('eventIgnitionOff')}</MenuItem>
		</Select>
	 </FormControl>
  )
}

tananaev replied:
I think the right solution is to provide a way for each report to add custom fields in addition to base ones.

tananaev replied:
I've added `FilterForm` that you can use here.

tananaev commented on modern/src/reports/EventReportPage.js:
```diff
@@ -0,0 +1,85 @@
+  }
+
+  return (
+    <div className={classes.root}>
```
There has to be some code here we can unify. For example top level layout and styles.

mail2bishnoi replied:
I'll do some work around this.

tananaev commented on modern/src/reports/EventReportPage.js:
```diff
@@ -0,0 +1,85 @@
+      to: to.toISOString(),
+    });
+
+    fetch(`/api/reports/events?${query.toString()}`, { headers: { Accept: 'application/json' } })
```
Use async/await.

mail2bishnoi replied:
done.

tananaev commented:
Any updates here? The reason I'm asking is that we plan to do some development around reports and don't want to add more merge conflicts.

mail2bishnoi commented:
> 
> 
> Any updates here? The reason I'm asking is that we plan to do some development around reports and don't want to add more merge conflicts.

I have updated the route report to use ReportFilter component. Also, I have added the EventType custom field to Events report. 

I have two approaches in my mind for providing custom fields for every report-
1. Pass a reportType attribute to ReportFilter and on the basis of reportType attribute add custom fields. (which I am using now)
2. Add custom fields as child of ReportFilter and in ReportFilter component use {props.children}.

Can you please suggest which aproach I should use.

I am actively working on this project and want to contribute.

tananaev commented:
I think second approach is better, thanks.

mail2bishnoi commented:
> 
> 
> I think second approach is better, thanks.

I have updated the ReportFilter, RouteReportPage, and EventReportPage using this approach. Please review them and suggest for the improvements. If that is fine then I will start working on some unification of the Report Layout.

tananaev commented on modern/.env:
```diff
@@ -1,2 +1,2 @@
 SKIP_PREFLIGHT_CHECK=true
-REACT_APP_URL_NAME='localhost:8082'
+REACT_APP_URL_NAME='13.235.202.9:8082'
```
Don't forget to revert this.

tananaev commented:
Looks pretty good. I think we can move to layout unification.

mail2bishnoi commented:
> 
> 
> Looks pretty good. I think we can move to layout unification.

I have done a small workaround on the report layout, please review and suggest improvements. I have defined the layout in a separate component "ReportView", and call this component from the respective report component.

tananaev commented on modern/src/reports/RouteReportPage.js:
```diff
@@ -1,85 +1,59 @@
-      from: from.toISOString(),
-      to: to.toISOString(),
-    });
+  const ReportFilterForm = ({ onResult }) => {
```
You can probably move this on the top level outside of `RouteReportPage`.

tananaev commented on modern/src/reports/RouteReportPage.js:
```diff
@@ -1,85 +1,59 @@
+    return <ReportFilter handleSubmit={handleSubmit} />;
+  }
+
+  const ReportListView = ({items}) => {
```
Instead of this let's put it as child of `ReportView`

tananaev commented on modern/src/reports/EventReportPage.js:
```diff
@@ -1,111 +1,88 @@
-    });
-    eventType.map(t=>query.append('type',t));
-    const response = await fetch(`/api/reports/events?${query.toString()}`, { headers: { Accept: 'application/json' } })
+  const ReportFilterForm = ({ onResult }) => {
```
Same for this.

mail2bishnoi replied:
Changes done. Please review them. Also changed one file name from "ReportView" to "ReportLayoutPage".

tananaev reviewed (commented):
Added some minor comments, but overall makes sense.

tananaev commented:
Not all comments are addressed. Also, please fix merge conflict. After that we can do final review.

mail2bishnoi commented:
> 
> 
> Not all comments are addressed. Also, please fix merge conflict. After that we can do final review.

I have fixed the merge conflicts. Which comments are left out?

tananaev commented:
Merged. Thanks for fixing all the comments.
